model/patch_lifecycle.go

```diff
@@ -143,28 +143,33 @@ func GetPatchedProject(ctx context.Context, p *patch.Patch, githubOauthToken str
 		hash = p.GithubPatchData.MergeCommitSHA
 	}
 
+	var path = projectRef.RemotePath
+	if p.Path != "" {
+		path = p.Path
+	}
+
 	githubFile, err := thirdparty.GetGithubFile(ctx, githubOauthToken, projectRef.Owner,
-		projectRef.Repo, projectRef.RemotePath, hash)
+		projectRef.Repo, path, hash)
 	if err != nil {
 		// if the project file doesn't exist, but our patch includes a project file,
 		// we try to apply the diff and proceed.
-		if !(p.ConfigChanged(projectRef.RemotePath) && thirdparty.IsFileNotFound(err)) {
+		if !(p.ConfigChanged(path) && thirdparty.IsFileNotFound(err)) {
 			// return an error if the github error is network/auth-related or we aren't patching the config
 			return nil, "", errors.Wrapf(err, "Could not get github file at '%s/%s'@%s: %s", projectRef.Owner,
-				projectRef.Repo, projectRef.RemotePath, hash)
+				projectRef.Repo, path, hash)
 		}
 	} else {
 		// we successfully got the project file in base64, so we decode it
 		projectFileBytes, err = base64.StdEncoding.DecodeString(*githubFile.Content)
 		if err != nil {
 			return nil, "", errors.Wrapf(err, "Could not decode github file at '%s/%s'@%s: %s", projectRef.Owner,
-				projectRef.Repo, projectRef.RemotePath, hash)
+				projectRef.Repo, path, hash)
 		}
 	}
 
 	// apply remote configuration patch if needed
-	if !(p.IsGithubPRPatch() || p.IsPRMergePatch()) && p.ConfigChanged(projectRef.RemotePath) {
-		projectFileBytes, err = MakePatchedConfig(ctx, env, p, projectRef.RemotePath, string(projectFileBytes))
+	if !(p.IsGithubPRPatch() || p.IsPRMergePatch()) && p.ConfigChanged(path) {
```

Comment:
How do we want to handle commit queue patches? Do we want to let users specify a path?

Reply:
Perhaps we would want to prevent this functionality for commit queue patches to protect against users from pointing to a new config file without any tests.  Alternatively we could keep the functionality and alert users of the implications of using a custom config for commit queue patches.  @ablack12  what's your perspective?

Reply:
Commit queue patches are actually created with a different CLI command (evergreen commit-queue merge) so they wouldn't ever have the option to change the path; that being said, it does seem reasonable to verify that it isn't a PR patch before setting path to the patch variable, just in case.

Reply:
That makes sense - changed such that path for a patch can only be changed if it isn't a PR patch and a non empty string is passed in through the CLI

Reply:
we may also want to explicitly verify the commit queue case (i.e. !p.isMergePatch)